com.unity.multiplayer.mlapi/Runtime/Core/SnapshotSystem.cs

```diff
@@ -183,14 +183,22 @@ public void NetworkUpdate(NetworkUpdateStage updateStage)
         private void SendSnapshot(ulong clientId)
         {
             // Send the entry index and the buffer where the variables are serialized
-            using (var buffer = PooledNetworkBuffer.Get())
-            {
-                WriteIndex(buffer);
-                WriteBuffer(buffer);
 
-                m_NetworkManager.MessageSender.Send(clientId, NetworkConstants.SNAPSHOT_DATA,
-                    NetworkChannel.SnapshotExchange, buffer);
-                buffer.Dispose();
+            var context = m_NetworkManager.MessageQueueContainer.EnterInternalCommandContext(
+                MessageQueueContainer.MessageType.SnapshotData,
+                NetworkChannel.SnapshotExchange,
+                new[] {clientId},
+                NetworkUpdateLoop.UpdateStage
+            );
+
+            if (context != null)
+            {
+                using (var icontext = (InternalCommandContext) context)
+                {
+                    var buffer = icontext.NetworkWriter.GetStream() as NetworkBuffer;
```

Comment:
```suggestion
                    var buffer = (NetworkBuffer) icontext.NetworkWriter.GetStream();
```

Reply:
Unless there's some hidden reason to want to have it be possible to be cast to `null` for whatever reason.

Reply:
Nope, this is me being new to C# and not realizing there's a difference between the two constructs.

Reply:
Done.